### DISCUSSION TITLE
Cheer me up guys

### DISCUSSION THREAD

**Discussion by imgazi** [Cyfrin/foundry-full-course-cu#2416] [Q&A]
I am new here. Previously I did patricks course on youtube but couldn't finish it cz i felt left out. It seems like i am the only one who is learning. So guys motivate me . I am starting all over again. This time on Cyfrin. 
And if there is any community please add me.

**Comment by huleimu** [comment]
Don’t worry, give it your all!

**Reply by DanCsoftware** [reply]
Hey! Just wanted to offer a bit of encouragement, it's okay if you don't understand something right away; it's also okay if your day doesn't include 3-5 hours of deep diving into the code course. (I think Patrick even dissuades people from doing this). 

I want to state that Patrick gives some solid recommendations on how to go about the course, deep diving into AI buddies, utilizing github discussion, etc., but here are just some of my opinions! 

1. Try and just be consistent, 30 minutes - 1 hour each day, maybe more if you're feeling like you're on a roll. 
2. Don't be afraid to pause the video, go back, maybe set the speed to 0.5 to clearly grasp whats going on.
3. Open up some Google Docs and start taking notes to explain the things you've learned or re-explaining the code that you written, when you begin a lesson on a day, write down a new date! This process should feel like a journal, record your progress and record your thought process. 
4. I personally have found a lot of encouragement from AI buddies, using those notes from the beginning of the year till now, on days I feel especially confused or stuck or maybe even needing some encouragement, I've fed those journal notes into the AI buddy and ask how my progress has been and you'd be surprised how a little each day eventually compounds and makes certain concepts more clear, concepts you found confusing back then, aren't so much anymore because that consistent behavior really does play a role, remember this is a language, not rocket science, so you need to always be interacting and facing that language. 
5. Question everything, if something doesn't make sense, explore that. Who cares if that means you'll take longer to finish the course, like question why we even write SPDX-License-Identifier at the top, or why in our import addresses we write our remappings in the foundry.toml, they're good habits to build and ultimately will make you holistically understand what's going on under the hood, if we don't do that we're just following along a guy on youtube typing funny named things. 

Hope this helps even a little bit! Onwards and upwards my friend.

**Reply by EngrPips** [reply]
Well said @DanCsoftware!

**Reply by imgazi** [reply]
@DanCsoftware thanks for those precious words . But i think this course is not updated anymore. I mean the webpage he shows for metamask and chainlink are almost different when i search them . it takes more time to understand whats gooing on

**Reply by EngrPips** [reply]
Well, if you are concerned about the codebase and the programming aspect of the course rather than the webpage of metamask and chainlink, then I will say the course is updated. You should take the course on cyfrin updraft as well https://updraft.cyfrin.io/ things get updated time to time on updraft than on youtube.

**Comment by EngrPips** [comment]
Hello @imgazi, welcome aboard. We wish you the best in your journey. Please join the official discord of Cyfrin https://discord.gg/t8qVHJCb. You will get more help and support on there.
